ui/src/utils/plugins.js

```diff
@@ -484,6 +484,22 @@ export const genericUtilPlugin = {
       const regexExp = /^[0-9a-fA-F]{8}\b-[0-9a-fA-F]{4}\b-[0-9a-fA-F]{4}\b-[0-9a-fA-F]{4}\b-[0-9a-fA-F]{12}$/gi
       return regexExp.test(uuid)
     }
+
+    app.config.globalProperties.$toBase64Encoded = function (text) {
+      const base64regex = /^([0-9a-zA-Z+/]{4})*(([0-9a-zA-Z+/]{2}==)|([0-9a-zA-Z+/]{3}=))?$/
+      if (base64regex.test(text)) {
+        return text
+      }
+      return btoa(unescape(encodeURIComponent(text)))
+    }
+
+    app.config.globalProperties.$toBase64AndURIEncoded = function (text) {
+      const base64regex = /^([0-9a-zA-Z+/]{4})*(([0-9a-zA-Z+/]{2}==)|([0-9a-zA-Z+/]{3}=))?$/
+      if (base64regex.test(text)) {
```

Comment:
This can be useful when user enters an already base64 encoded string.
Use-case: cloud-init allows passing gzipped content. To pass it ACS API user will already have a base64 encoded string created with means like, https://base64.guru/converter/encode/file

Reply:
@shwstppr 
if user input a base64-encoded string, it will be passed to cloudstack API without encoding again, right ?
what if user wants to use the base64-encoded string as the userdata ?

Reply:
@weizhouapache why would user want to pass a double base64 encoded string?
cloud-init doesn't have a data format for that, https://cloudinit.readthedocs.io/en/latest/explanation/format.html
Even if user wants that in some scenarios it can be done manually.

Reply:
> @weizhouapache why would user want to pass a double base64 encoded string? cloud-init doesn't have a data format for that, https://cloudinit.readthedocs.io/en/latest/explanation/format.html Even if user wants that in some scenarios it can be done manually.

@shwstppr 
cloud-init is a way to use userdata, but not the only one.
maybe we can add a checkbox to indicate if the input is base64-encoded or not. we can do it later.